osxkeychain.go

```diff
@@ -176,43 +188,61 @@ func AddInternetPassword(pass *InternetPassword) error {
 //
 // Returns an error if the lookup was unsuccessful.
 func FindInternetPassword(pass *InternetPassword) (*InternetPassword, error) {
-	resp := InternetPassword{}
-	protocol := protocolTypeToC(pass.Protocol)
-	authtype := C.SecAuthenticationType(C.kSecAuthenticationTypeHTTPBasic)
-	var cpassword unsafe.Pointer
-	var cpasslen C.UInt32
+	// TODO: Check fields for UTF-8 encoding and size fitting in
+	// 32 bits.
+
+	serverName := C.CString(pass.ServerName)
+	defer C.free(unsafe.Pointer(serverName))
+
+	// TODO: Make optional.
+	securityDomain := C.CString(pass.SecurityDomain)
+	defer C.free(unsafe.Pointer(securityDomain))
+
+	accountName := C.CString(pass.AccountName)
+	defer C.free(unsafe.Pointer(accountName))
+
+	path := C.CString(pass.Path)
+	defer C.free(unsafe.Pointer(path))
+
+	protocol := C.uint(protocolTypeToC(pass.Protocol))
+
+	authtype := C.uint(authenticationTypeToC(pass.AuthType))
+
+	var passwordLength C.UInt32
+
+	var password unsafe.Pointer
+
```

Comment:
No need for the newlines between `protocol := ..` and here, they're all just simple variable declarations.